include/fmt/format.h

```diff
@@ -914,17 +910,17 @@ class basic_memory_buffer : public detail::buffer<T> {
   static FMT_CONSTEXPR20 void grow(detail::buffer<T>& buf, size_t size) {
     detail::abort_fuzzing_if(size > 5000);
     auto& self = static_cast<basic_memory_buffer&>(buf);
-    constexpr size_t max_size =
-        detail::max_value<typename detail::allocator_size<Allocator>::type>() /
-        sizeof(T);
+    const size_t max_size =
+        std::allocator_traits<Allocator>::max_size(self.alloc_);
     size_t old_capacity = buf.capacity();
     size_t new_capacity = old_capacity + old_capacity / 2;
     if (size > new_capacity)
       new_capacity = size;
     else if (new_capacity > max_size)
       new_capacity = size > max_size ? size : max_size;
     T* old_data = buf.data();
-    T* new_data = self.alloc_.allocate(new_capacity);
+    T* new_data =
+        std::allocator_traits<Allocator>::allocate(self.alloc_, new_capacity);
```

Comment:
I don't think we need to use traits here and in deallocate below. It's mainly useful for max_size and if we used a hint (which we don't).

Reply:
Ok.